### PR TITLE
RavenDB-17130 fixed issue with IfEntityIs returning always null in index

### PR DESCRIPTION
**Pull request by ppekrol**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-17130

### Additional description

`IfEnitityIs` was not invoked by the index properly, yielding null. Removed `MetadataExtensions` because it is no longer needed. `WhereEntityIs` is handled by collection name retriever so there is no invocation there.

### Type of change

- Bug fix

### How risky is the change?

- Low 

### Backward compatibility

- Non breaking change

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing 

- Tests have been added that prove the fix is effective or that the feature works

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

File: src/Raven.Server/Documents/Indexes/Static/Extensions/DynamicExtensionMethods.cs [hunk 1]
```diff
@@ -15,6 +15,19 @@ public static BoostedValue Boost(dynamic o, object value)
             };
         }
 
+        public static dynamic IfEntityIs(dynamic o, string collection)
```
Comment (ayende) [inline src/Raven.Server/Documents/Indexes/Static/Extensions/DynamicExtensionMethods.cs:18]: don't we need this as an array as well?

File: src/Raven.Server/Documents/Indexes/Static/Extensions/MetadataExtensions.cs [hunk 2]
```diff
@@ -1,38 +0,0 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-
-namespace Raven.Server.Documents.Indexes.Static.Extensions
-{
-    public static class MetadataExtensions
-    {
-        public static IEnumerable<dynamic> WhereEntityIs(this IEnumerable<dynamic> self, params string[] collections)
```
Comment (ayende) [inline src/Raven.Server/Documents/Indexes/Static/Extensions/MetadataExtensions.cs:9]: to match this?

Comment (ppekrol) [reply src/Raven.Server/Documents/Indexes/Static/Extensions/MetadataExtensions.cs:9]: I think that those are two different things. `WhereEntityIs` is filtering the collections on which the index is working and `IfEnitytIs` is useful when doing `let x = ` to get appropriate documents in there.

Comment (ayende) [reply src/Raven.Server/Documents/Indexes/Static/Extensions/MetadataExtensions.cs:9]: IIRC, that was to be able to do something like `docs.WhereEntityIs("Abc", "Def", etc)`

Comment (ppekrol) [reply src/Raven.Server/Documents/Indexes/Static/Extensions/MetadataExtensions.cs:9]: I do not follow. What's the purpose of `IfEntityIs` then?

Comment (ppekrol) [reply src/Raven.Server/Documents/Indexes/Static/Extensions/MetadataExtensions.cs:9]:
```
IEnumerable<TResult> WhereEntityIs<TResult>(this IEnumerable<object> queryable, params string[] names)

TResult IfEntityIs<TResult>(this object queryable, string name)
```

notice the difference. This is why `WhereEntityIs` is good at `from doc in docs.WhereEntityIs()` and the other one IMO for
```
from doc in docs.WhereEntityIs("Dogs", "Cats")
let dog = doc.IfEntityIs("Dogs")
let cat = doc.IfEntityIs("Cats")
select ...
```

Comment (ppekrol) [reply src/Raven.Server/Documents/Indexes/Static/Extensions/MetadataExtensions.cs:9]: And we can add `params` to `IfEntityIs` although I'm not sure if this is that usable.

Comment (ayende) [reply src/Raven.Server/Documents/Indexes/Static/Extensions/MetadataExtensions.cs:9]: OK

**Pull request merged**